src/Network/ServerHandler.cpp

```diff
@@ -0,0 +1,142 @@
+#include "EwECS/Network/ServerHandler.hpp"
+#include <boost/bind.hpp>
+#include <iostream>
+#include "EwECS/Event/EventManager.hpp"
+#include "EwECS/Network/NetworkHandler.hpp"
+#include "EwECS/Network/Packet.hpp"
+#include "EwECS/Network/Values.hpp"
+
+namespace ECS::Network {
+
+    using boost::asio::ip::udp;
+
+    void ServerHandler::start(std::string &aHost, unsigned short aPort, unsigned short aMaxClients,
+                              PacketFactory &aPacketFactory)
+    {
+        NetworkHandler &network = NetworkHandler::getInstance();
+        udp::endpoint endpoint(boost::asio::ip::address::from_string(aHost), aPort);
+
+        network.onReceive([this](int8_t aType, IPayload *aPayload, udp::endpoint &aClientEndpoint) {
+            if (aType == AKNOWLEDGMENT_PACKET_TYPE) {
+                handleAknowledgment(aClientEndpoint);
+            } else if (aType == ERROR_PACKET_TYPE) {
+                // receive error packet
```

Comment:
Commented if ?

Reply:
i'll add a errorCallback to server and network handlers even tough they won't be used by rtype